File: java/RemoveOccurence.java

public class RemoveOccurence {

    public static int removeOccurence(int[] nums, int target) {
        int left = 0;

        for (int right = 0; right < nums.length; right++) {
            if (nums[right] != target) {
                nums[left] = nums[right];
                left ++;
            }
        }
        return left;
    }

    public static void main(String[] args) {
        int[] numsArr = {3, 2, 2, 3, 4, 3, 5};
        int targetNum = 3;
        int newLength = removeOccurence(numsArr, targetNum);
        System.out.println("New length: " + newLength);

        for(int i = 0; i < newLength; i++){
            System.out.print(numsArr[i] + " ");
        }

    }
}

File: typescript/RemoveOccurence.ts
/* 
https://www.geeksforgeeks.org/dsa/remove-element/
*/

const numsArr: number[] = [3, 2, 2, 3, 4, 3, 5];
const targetNums: number = 3;

function RemoveOccurence(nums: number[], target: number): number {
  let left: number = 0;
  let count: number = 0;

  for (let right = 0; right < nums.length; right++) {
    console.log("ITERATION ", count++);
    console.log(`left[${left}] = `, nums[left]);
    console.log(`right[${right}] = `, nums[right]);
    console.log("nums array:", nums);
    console.log("_______________");

    if (nums[right] !== target) {
      
      nums[left] = nums[right];
      left++;
    }
  }
  return left;
}

console.log(RemoveOccurence(numsArr, targetNums))
